lib/floe/workflow/state.rb

```diff
@@ -19,27 +19,40 @@ def build!(workflow, name, payload)
         end
       end
 
-      attr_reader :workflow, :comment, :name, :type, :payload
+      attr_reader :workflow, :comment, :name, :next, :type, :payload
 
       def initialize(workflow, name, payload)
         @workflow = workflow
         @name     = name
         @payload  = payload
-        @end      = !!payload["End"]
         @type     = payload["Type"]
         @comment  = payload["Comment"]
+        # some nodes like Fail, and Succeed have a hardcoded next of nil
+        # Choice ignores this (and has a default for their static fallback state)
+        @next     = payload["End"] ? nil : payload["Next"]
+        # all nodes but Fail use this:
+        @input_path  = Path.new(payload.fetch("InputPath", "$"))
+        @output_path = Path.new(payload.fetch("OutputPath", "$"))
```

Comment:
The fact that Fail doesn't have `@input_path` and `@output_path` defined is why we didn't want to handle this in the base `State` class.  It doesn't make sense for `Fail` to have these ivars